File: 백준/Silver/10845. 큐/큐.py
from collections import deque
import sys

N = int(input())
deque = deque()

def play(command, num):
    if command == "push":
        deque.append(num)
    elif command == "pop":
        if len(deque) == 0:
            print(-1)
        else:
            print(deque.popleft())
    elif command == "size":
        print(len(deque))
    elif command == "empty":
        if len(deque) == 0:
            print(1)
        else:
            print(0)
    elif command == "front":
        if len(deque) == 0:
            print(-1)
        else:
            print(deque[0])
    elif command == "back":
        if len(deque) == 0:
            print(-1)
        else:
            print(deque[-1])

for _ in range(N):
    input_data = list(map(str, sys.stdin.readline().strip().split(' ')))
    command = input_data[0]
    num = 0
    if len(input_data) > 1:
        num = int(input_data[1])
    play(command, num)

File: 백준/Silver/10845. 큐/큐.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		final int N = Integer.parseInt(br.readLine());
		StringBuilder sb = new StringBuilder();
		Queue<Integer> que = new LinkedList<>();
		// poll() : 헤드 요소를 조회(출력 가능)하고 제거, 하지만 큐가 비어 있다면 null 반환
		// peek() : 헤드 요소 조회 및 반환, 하지만 큐가 비어 있다면 null 반환
		int back = 0;
		for (int i = 0; i < N; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			String order = st.nextToken();
			if (order.equals("push")) {
				int num = Integer.parseInt(st.nextToken());
				back = num;
				que.add(num);
			} else if (order.equals("pop")) {
				if (que.isEmpty()) {
					sb.append(-1).append('\n');
				} else {
					sb.append(que.poll()).append('\n');
				}
			} else if (order.equals("size")) {
				sb.append(que.size()).append('\n');
			} else if (order.equals("empty")) {
				if (que.isEmpty()) {
					sb.append(1).append('\n');
				} else {
					sb.append(0).append('\n');
				}
			} else if (order.equals("front")) {
				if (que.isEmpty()) {
					sb.append(-1).append('\n');
				} else {
					sb.append(que.peek()).append('\n');
				}
			} else if (order.equals("back")) {
				if (que.isEmpty()) {
					sb.append(-1).append('\n');
				} else {
					sb.append(back).append('\n');
				}
			}
		}
		System.out.println(sb);
	}
}